### ISSUE TITLE
OptionHash should accept reasonable usages

### ISSUE THREAD

**Issue by swordray** [bbatsov/rubocop#5652]
This is a case that override Rails `image_tag` for lazyload js. There is no need to expand options to keyword arguments. And the original method signature is exectly [image_tag(source, options = {})](http://api.rubyonrails.org/classes/ActionView/Helpers/AssetTagHelper.html#method-i-image_tag).

```ruby
module ApplicationHelper
  def image_tag(source, options = {})
    options['data-src'] = image_path(source).presence
    source = 'data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=='
    super
  end
end
```

So should we check if `options` is used only accessing values or not?

## Expected behavior

Don't flag an offense.

## Actual behavior

Flag an offense.

## Steps to reproduce the problem

```yaml
Style/OptionHash:
  Enabled: true
```

## RuboCop version

```
$ rubocop -V
0.53.0 (using Parser 2.5.0.2, running on ruby 2.5.0 x86_64-darwin17)
```

**Comment by Drenmi** [comment]
RuboCop isn't capable of reason, so it can't really judge what is reasonable. 🙂 

However, I think your example is valuable, because it calls `super`, which will pass the arguments on, indicating that the hash is expected. We could certainly account for this case.
